src/PostgREST/ApiRequest/Preferences.hs

```diff
@@ -216,12 +216,14 @@ instance ToAppliedHeader PreferTransaction
 -- How to handle the insertion/update when the keys specified in ?columns are not present
 -- in the json body.
 data PreferUndefinedKeys
-  = ApplyDefaults  -- ^ Use the default column value for the unspecified keys.
-  | IgnoreDefaults -- ^ Inserts: null values / Updates: the keys are not SET to any value
+  = ApplyDefaults  -- ^ Inserts and Updates: Use the default column value for the unspecified keys.
+  | IgnoreValues   -- ^ Inserts: null values | Updates: the keys are not SET to any value
+  | ApplyNulls     -- ^ Inserts: null values | Updates: SET the values to null
```

Comment:
I noticed that the comment I added in the other PR is wrong. 

```haskell
| IgnoreDefaults -- ^ Inserts: null values / Updates: the keys are not SET to any value
```

When no header is sent, it updates applying `null` values instead of "ignoring" the key and keeping the old value. 

I added other header as a proposal:
- `IgnoreDefaults` is now `ApplyNulls` and is the default behavior when no header is sent, that is, it updates the unspecified columns to `null` values.
- `IgnoreValues` does not set to any value (in reality it sets to the original value but the result is the same).